### DISCUSSION TITLE
/api/pipelines/list 500

### DISCUSSION THREAD

**Discussion by openainext** [open-webui/pipelines#51] [General]
![image](https://github.com/open-webui/pipelines/assets/456516/839472f4-234c-4d78-8028-912c8bf47845)
/api/pipelines/list 500

**Comment by melvincarvalho** [comment]
I tried this:

![image](https://github.com/open-webui/pipelines/assets/65864/d7e37809-c379-4d14-9279-5fadcf17d4cd)

but I didnt see pipelines in the Admin panel

![image](https://github.com/open-webui/pipelines/assets/65864/c62bcc9c-1b65-4331-8aac-c4d031ac85a7)

Going here on loncalost:

> /api/pipelines/list

Gives:

> {"detail":"Not authenticated"}

But I'm signed in

**Comment by tjbck** [comment]
What method did you use to install Pipelines? if they're both on docker you should use `host.docker.internal` instead of `localhost`. Keep us updated!

**Reply by kartguru** [reply]
thanks @tjbck

**Comment by melvincarvalho** [comment]
Yes, both docker, I will debug, and report back, thanks!

**Comment by ther3zz** [comment]
@melvincarvalho 
Wondering if this one is related: https://github.com/open-webui/open-webui/issues/2723

I'm betting that if you refresh the page, your UI breaks and doesn't load.

If that's the case, just modify the config.json of open webui, something like this:

```
	"openai": {
		"api_base_urls": [
			"http://192.168.1.200:4000/v1",
			"http://192.168.1.200:9099"
		],
		"api_keys": [
			"sk-xxx",
			"0p3n-w3bu!"
		],
		"enable": true
	},
```

In this example, the first IP would be my LiteLLM instance and second IP is the piplines instance. The first api_key is for litellm and second one is for pipelines.

**Comment by melvincarvalho** [comment]
My 2 commands

## open-webui

> run -d -p 6111:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main

## pipelines

> docker run -d -p 9099:9099 --add-host=host.docker.internal:host-gateway -v pipelines:/app/pipelines --name pipelines --restart always ghcr.io/open-webui/pipelines:main

I figured out where the pipelines tab is, but it's empty:

![image](https://github.com/open-webui/pipelines/assets/65864/d7fb8ce1-a97d-4c6a-99b7-6ce7140672d5)

**Comment by melvincarvalho** [comment]
## debug

When clicking refresh in connections it did give me this, although both dockers are running

![image](https://github.com/open-webui/pipelines/assets/65864/3bb2b1e7-655b-428e-a780-1fca28ec6c3b)

**Comment by timmytaco45** [comment]
<img width="574" alt="image" src="https://github.com/open-webui/pipelines/assets/171052332/1bcef1e2-4579-4b6e-b81b-d910fc373d83">

I get this when openai url is host.docker.internal:9099 and when it is localhost:9099 I get the 'openai: network problem'. I have never successfully used a pipe, filter, or anything without error.

**Comment by tjbck** [comment]
@timmytaco45 Did you install any pipelines?

**Comment by timmytaco45** [comment]
I added the urls for 'function calling python' and 'applescript'. They were visible but neither worked and broke normal chat use so I assumed the /examples folder was being sorted out. I will try again.

**Reply by rootcli** [reply]
Try the Wikipedia Pipeline

**Comment by jkirk-denaliai** [comment]
I am using only the wikipedia pipeline for testing.
When I navigate to http://localhost:9099/models I see this:

`{
    "data": [
        {
            "id": "wikipedia_pipeline",
            "name": "Wikipedia Pipeline",
            "object": "model",
            "created": 1718755863,
            "owned_by": "openai",
            "pipeline": {
                "type": "pipe",
                "valves": true
            }
        }
    ],
    "object": "list",
    "pipelines": true
}`

When I use these settings under connections:
http://localhost:9099
0p3n-w3bu!

I get a "OpenAI: Network Problem" error

I'm on MacOS, Python 3.12, running pipelines locally, not in docker.

**Reply by jkirk-denaliai** [reply]
Okay open-webui is in docker.  So localhost is local to docker.  I had to bind pipelines to my actual IP and use that IP.
